src/magics.c

```diff
@@ -109,309 +64,96 @@ const uint64_t MagicNumberBishop[SQUARE_NB] = {
     0x0400000260142410ull, 0x0800633408100500ull, 0xfc087e8e4bb2f736ull, 0x43ff9e4ef4ca2c89ull
 };
 
-static void setSquare(uint64_t *bb, int r, int f) {
-    if (0 <= r && r < RANK_NB && 0 <= f && f < FILE_NB)
-        setBit(bb, square(r, f));
-}
+static uint64_t RookAttacks[0x19000], BishopAttacks[0x1480];
+static uint64_t *BishopAttacksPtr[SQUARE_NB], *RookAttacksPtr[SQUARE_NB];
 
-static void generateLeaperAttacks() {
+static uint64_t BishopMask[SQUARE_NB], RookMask[SQUARE_NB];
+static unsigned BishopShift[SQUARE_NB], RookShift[SQUARE_NB];
 
-    const int KnightLeaps[8][2] = {{-2,-1}, {-2,1}, {-1,-2}, {-1,2}, {1,-2}, {1,2}, {2,-1}, {2,1}};
-    const int KingLeaps[8][2] = {{-1,-1}, {-1,0}, {-1,1}, {0,-1}, {0,1}, {1,-1}, {1,0}, {1,1}};
+static uint64_t sliderAttacks(int s, uint64_t occ, const int dir[4][2]) {
 
-    for (int s = 0; s < SQUARE_NB; s++) {
-        const int r = rankOf(s), f = fileOf(s);
+    uint64_t result = 0;
 
-        for (int d = 0; d < 8; d++) {
-            setSquare(&KnightAttacks[s], r + KnightLeaps[d][0], f + KnightLeaps[d][1]);
-            setSquare(&KingAttacks[s], r + KingLeaps[d][0], f + KingLeaps[d][1]);
-        }
-    }
-}
+    for (int i = 0; i < 4; i++) {
+        int dr = dir[i][0], df = dir[i][1];
+        int r, f;
 
-static void generateRookIndexes(){
-    
-    int i, sum;
-    
-    for (i = 0, sum = 0; i < SQUARE_NB; i++){
-        MagicRookIndexes[i] = sum;
-        sum += (1 << (64 - MagicShiftsRook[i]));
-    }
-}
+        for (r = rankOf(s) + dr, f = fileOf(s) + df;
+                0 <= r && r < RANK_NB && 0 <= f && f < FILE_NB;
```

Comment:
Line up the indentation for each part of the for loop?

Reply:
I know this may seem unusual but there's a readability trade-off here. If you indent only once, it's on the same level as the body of the loop, which is visually confusing. If you don't indent and leave it at the same level as the for, it's an eye sore.

It's very rare anyway to have for loops like that with the condition on another line (because it's too complex to fit in one line). So I wondered how to format it. I checked with my favorite formatter `astyle` and that's what it did.